software/firmware/europi.py

```diff
@@ -1,580 +1,619 @@
-        padding_top = (self.height - (len(lines) * 9)) / 2
-        for index, content in enumerate(lines):
-            x_offset = int((self.width - ((len(content) + 1) * 7)) / 2) - 1
-            y_offset = int((index * 9) + padding_top) - 1
-            self.text(content, x_offset, y_offset)
-        self.show()
-
-
-class Output:
-    """A class for sending digital or analogue voltage to an output jack.
-
-    The outputs are capable of providing 0-10V, which can be achieved using
-    the ``cvx.voltage()`` method.
-
-    So that there is no chance of not having the full range, the chosen
-    resistor values actually give you a range of about 0-10.5V, which is why
-    calibration is important if you want to be able to output precise voltages.
-    """
-
-    def __init__(self, pin, min_voltage=MIN_OUTPUT_VOLTAGE, max_voltage=MAX_OUTPUT_VOLTAGE):
-        self.pin = PWM(Pin(pin))
-        self.pin.freq(PWM_FREQ)
-        self._duty = 0
-        self.MIN_VOLTAGE = min_voltage
-        self.MAX_VOLTAGE = max_voltage
-
-        self._gradients = []
-        for index, value in enumerate(OUTPUT_CALIBRATION_VALUES[:-1]):
-            self._gradients.append(OUTPUT_CALIBRATION_VALUES[index + 1] - value)
-        self._gradients.append(self._gradients[-1])
-
-    def _set_duty(self, cycle):
-        cycle = int(cycle)
-        self.pin.duty_u16(clamp(cycle, 0, MAX_UINT16))
-        self._duty = cycle
-
-    def voltage(self, voltage=None):
-        """Set the output voltage to the provided value within the range of 0 to 10."""
-        if voltage is None:
-            return self._duty / MAX_UINT16
-        voltage = clamp(voltage, self.MIN_VOLTAGE, self.MAX_VOLTAGE)
-        index = int(voltage // 1)
-        self._set_duty(OUTPUT_CALIBRATION_VALUES[index] + (self._gradients[index] * (voltage % 1)))
-
-    def on(self):
-        """Set the voltage HIGH at 5 volts."""
-        self.voltage(5)
-
-    def off(self):
-        """Set the voltage LOW at 0 volts."""
-        self._set_duty(0)
-
-    def toggle(self):
-        """Invert the Output's current state."""
-        if self._duty > 500:
-            self.off()
-        else:
-            self.on()
-
-    def value(self, value):
-        """Sets the output to 0V or 5V based on a binary input, 0 or 1."""
-        if value == HIGH:
-            self.on()
-        else:
-            self.off()
-
-
-## Initialize EuroPi global singleton instance variables.
-
-# Define all the I/O using the appropriate class and with the pins used
-din = DigitalInput(22)
-ain = AnalogueInput(26)
-k1 = Knob(27)
-k2 = Knob(28)
-b1 = Button(4)
-b2 = Button(5)
-
-oled = Display(0, 1)
-cv1 = Output(21)
-cv2 = Output(20)
-cv3 = Output(16)
-cv4 = Output(17)
-cv5 = Output(18)
-cv6 = Output(19)
-cvs = [cv1, cv2, cv3, cv4, cv5, cv6]
-
-usb_connected = DigitalReader(24, 0)
-
-# Overclock the Pico for improved performance.
-freq(OVERCLOCKED_CPU_FREQ)
-
-# Reset the module state upon import.
-reset_state()

```

Comment:
Some how github shows the whole file as changed in the div?
Maybe because of changed line ending as described [here](https://stackoverflow.com/questions/30752959/git-showing-entire-file-is-modified-instead-of-showing-modified-small-part-of-co)?

Reply:
I rebased my PR to get the last updates from the main branch. 
The line ending problem is unfortunately a common one. Maybe it's on my side because I'm using both a Mac and Linux box and, since I just replaced my Mac with a new Mac Mini, I forgot to check my git config on this new machine. 

We could delete this PR and I could re-create a new one. Maybe it's the simplest solution.